### PR TITLE
Add SparseVectorSerializer.

### PR DESCRIPTION
**Pull request by andrewsmartin**
This is still kind of WIP, just want to get some eyes on this. I think I will add some tests for this one since it's not as trivial.

### CODE REVIEW

File: scio-extra/src/main/scala/com/spotify/scio/extra/breeze/BreezeKryoRegistrar.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    k.register(classOf[SparseVector[_]], new SparseVectorSerializer[Any])
+  }
+}
+
+private class SparseVectorSerializer[T] extends KSerializer[SparseVector[T]] {
+  override def read(kryo: Kryo, input: Input, tpe: Class[SparseVector[T]]): SparseVector[T] = {
+    val zero = kryo.readClassAndObject(input).asInstanceOf[T]
+    val len = input.read()
+    val data = new mutable.ArrayBuffer[T](len)
+    val index = new mutable.ArrayBuffer[Int](len)
```
Comment (richwhitjr) [inline scio-extra/src/main/scala/com/spotify/scio/extra/breeze/BreezeKryoRegistrar.scala]: Can this just be any Array instead since we know the length at creation time?  Could save some overhead since we don't have to worry about it growing.

Comment (andrewsmartin) [reply scio-extra/src/main/scala/com/spotify/scio/extra/breeze/BreezeKryoRegistrar.scala]: True, will update to just use plain `Array`.

File: build.sbt [hunk 2]
```diff
@@ -434,7 +434,7 @@ lazy val scioExtra: Project = Project(
   "scio-extra",
   file("scio-extra")
 ).settings(
-  commonSettings ++ itSettings,
+  commonSettings ++ itSettings ++ macroSettings,
```
Comment (nevillelyh) [inline build.sbt]: Why is this needed?

File: scio-extra/src/main/scala/com/spotify/scio/extra/breeze/breeze.scala [hunk 3]
```diff
@@ -17,8 +17,8 @@
 
 package com.spotify.scio.extra
 
-import breeze.linalg.operators.OpAdd
-import breeze.linalg.support.CanCopy
+import _root_.breeze.linalg.operators.OpAdd
```
Comment (nevillelyh) [inline scio-extra/src/main/scala/com/spotify/scio/extra/breeze/breeze.scala]: Why do we need to prefix all these imports?

File: build.sbt (hunk 2 above)
Comment (andrewsmartin) [reply build.sbt]: For the `@KryoRegistrar` macro.

File: scio-extra/src/main/scala/com/spotify/scio/extra/breeze/breeze.scala (hunk 3 above)
Comment (andrewsmartin) [reply scio-extra/src/main/scala/com/spotify/scio/extra/breeze/breeze.scala]: Because I moved the existing Breeze stuff into its own `breeze` package, which collides with the root namespace.

**Comment by andrewsmartin** [comment]
Gonna close this one as it doesn't handle specialized types properly. I'll re-open another one with some test cases when it's ready.
